### PR TITLE
Improve docs to emphasize remark-slug is required

### PR DESCRIPTION
**Pull request by lawrencecchen**
I was really confused until I saw #4. I don't think the current docs are clear enough. Please consider this fix!

### CODE REVIEW

**Comment by wooorm** [comment]
I dunno, I think this new note is misleading: there are other ways to add IDs (e.g., manually, different tools).

Perhaps the first description can be edited instead, something like:

```diff
-rehype plugin to automatically add links to headings (h1-h6).
+rehype plugin to automatically add links to headings (h1-h6) that already have an ID.
```

File: readme.md [hunk 1]
```diff
@@ -15,9 +15,11 @@
 [npm][]:
 
 ```sh
-npm install rehype-autolink-headings
+npm install rehype-autolink-headings rehype-slug
```
Comment (ChristianMurphy) [inline readme.md:18]:
```suggestion
npm install rehype-autolink-headings
```

`rehype-slug` is not needed, any method for adding ids to headings could be used.

File: readme.md [hunk 2]
```diff
@@ -15,9 +15,11 @@
...
 ```
 
+> **This package will not work without `rehype-slug`. More information under [API](#user-content-api).**
```
Comment (ChristianMurphy) [inline readme.md:21]: A more accurate version of this is noted in https://github.com/rehypejs/rehype-autolink-headings#rehypeuselink-options

If wanted this note could be moved up higher in the readme.

```suggestion
> Note: this plugin expects `id`s to already exist on headings. One way to add those automatically, is [rehype-slug](https://github.com/rehypejs/rehype-slug) (see example).
```

**Review by ChristianMurphy — CHANGES_REQUESTED** [review]

**Comment by wooorm** [comment]
I’ve pushed a change that hopefully improves this, but could not merge your PR as there are still unresolved suggestions.

**Pull request closed without merging**